syntax = "proto3";

// Get protoc here https://github.com/protocolbuffers/protobuf/releases
// .\protoc --python_out=. --pyi_out=. CAP.proto

message ActorInfo {
    string name = 1;
    optional string namespace = 2;
    optional string description = 3;
}

message ActorRegistration {
    ActorInfo actor_info = 1;
}

message ActorLookup {
    optional ActorInfo actor_info = 1;
    // TODO:  May need more structure here for symantic service discovery
    // optional string service_descriptor = 2;
}

